torch/nn/modules/activation.py

```diff
@@ -772,11 +789,26 @@ def forward(self, query, key, value, key_padding_mask=None,
         - attn_output_weights: :math:`(N, L, S)` where N is the batch size,
           L is the target sequence length, S is the source sequence length.
         """
-        return F.multi_head_attention_forward(
-            query, key, value, self.embed_dim, self.num_heads,
-            self.in_proj_weight, self.in_proj_bias, self.bias_k, self.bias_v, self.add_zero_attn,
-            self.dropout, self.out_proj.weight, self.out_proj.bias, training=self.training,
-            key_padding_mask=key_padding_mask, need_weights=need_weights, attn_mask=attn_mask)
+        if hasattr(self, '_qkv_same_embed_dim') and self._qkv_same_embed_dim is False:
```

Comment:
Why do you check it using an attribute?

Reply:
This will resolve the backward capability issue.

Reply:
I don't think this is necessary because we're updating the functional together with the module.

Reply:
some models trained by the old multiheadattention modules will break if there is not such a check, if my understanding is correct. Because they don't have q/k/v_proj_weight attributes.